waiter/src/waiter/simulator.clj

```diff
@@ -78,104 +87,129 @@
-                         :servers-started servers-started
-                         :completing-requests completing-requests
-                         :available-servers available-servers
-                         :eligible-servers eligible-servers
-                         :queued-clients queued-clients'
-                         :idle-servers idle-servers'
-                         :active-requests active-requests'
-                         :starting-servers starting-servers'
-                         :outstanding-requests (+ (count active-requests') queued-clients')
-                         :total-clients (+ (count active-requests') (count idle-clients') queued-clients')
-                         :total-instances total-instances
-                         :total-queue-time (+ total-queue-time queued-clients)
-                         :total-idle-server-time (+ total-idle-server-time idle-servers)
-                         :utilization utilization
-                         :healthy-instances (+ (count active-requests') idle-servers')
-                         :scale-amount scale-amount
-                         :target-instances target-instances
-                         :tick tick
-                         :scale-ups (+ scale-ups (if (pos? scale-amount) 1 0))
-                         :scale-downs (+ scale-downs (if (neg? scale-amount) 1 0)))))
+                      0.0)
+        waste (if (pos? total-instances)
+                (- 100.0 utilization)
+                0.0)]
+    (assoc current-state
+      :activating-clients activating-clients
+      :active-requests active-requests'
+      :available-servers available-servers
+      :clients-to-kill clients-to-kill
+      :completing-requests completing-requests
+      :eligible-servers eligible-servers
+      :healthy-instances (+ (count active-requests') idle-servers')
+      :idle-clients idle-clients'
+      :idle-servers idle-servers'
+      :interested-clients interested-clients
+      :outstanding-requests (+ (count active-requests') (count queued-clients''))
+      :queued-clients queued-clients''
+      :scale-amount scale-amount
+      :scale-ups (+ scale-ups (if (pos? scale-amount) 1 0))
+      :scale-downs (+ scale-downs (if (neg? scale-amount) 1 0))
+      :servers-started servers-started
+      :starting-servers starting-servers'
+      :target-instances target-instances
+      :tick tick
+      :total-clients (+ (count active-requests') (count idle-clients') (count queued-clients''))
+      :total-idle-server-time (+ total-idle-server-time idle-servers)
+      :total-instances total-instances
+      :total-queue-time (+ total-queue-time (count queued-clients))
+      :utilization utilization
+      :wait-time-max wait-time-max'
+      :wait-time-max-ema wait-time-max-ema'
+      :waste waste)))
 
-(let [default-initial-state {:idle-clients []
-                             :queued-clients 0
+(let [default-initial-state {:active-requests []
```

Comment:
Please keep formatting changes in separate PRs so it's straightforward to see what changed.